commit b7599e205eb559dff7ad0d62ac3ad5bb67060ec3
Author: Paul Wouters <paul@libreswan.org>
Date:   Mon Aug 20 21:44:53 2012 -0400

    * barf: don't search for our logs in lastlog, btmp or wtmp
    
    This is rhbz#https://bugzilla.redhat.com/show_bug.cgi?id=771612

diff --git a/programs/barf/barf.in b/programs/barf/barf.in
index 9798df4..2f4632a 100755
--- a/programs/barf/barf.in
+++ b/programs/barf/barf.in
@@ -72,7 +72,7 @@ findlog() {		# findlog string fallbackstring possiblefile ...
 		fi
 	done
 	# nope, resort to a search, newest first, of uncompressed logs
-	for f in `ls -t $LOGS | egrep -v '^mail' | egrep -v '\.(gz|Z)$'`
+	for f in `ls -t $LOGS | egrep -v lastlog | egrep -v tmp | egrep -v '^mail' | egrep -v '\.(gz|Z)$'`
 	do
 		if test -f $LOGS/$f -a ! -d $LOGS/$f && egrep -q "$s" $LOGS/$f
 		then
@@ -83,7 +83,7 @@ findlog() {		# findlog string fallbackstring possiblefile ...
 			return 0
 		fi
 	done
-	for f in `ls -t $LOGS | egrep -v '^mail' | egrep -v '\.(gz|Z)$'`
+	for f in `ls -t $LOGS | egrep -v lastlog | egrep -v tmp | egrep -v '^mail' | egrep -v '\.(gz|Z)$'`
 	do
 		if test -s $LOGS/$f -a -f $LOGS/$f && egrep -q "$t" $LOGS/$f
 		then
